assets/training/finetune_acft_hf_nlp/environments/acpt/context/finetune_run.py

```diff
@@ -429,15 +516,32 @@ def initiate_run():
 
     _initiate_run(completion_files_folder, model_selector_output,
                   preprocess_output, pytorch_model_folder, mlflow_model_folder)
-    cleanup(completion_files_folder, model_selector_output,
-            preprocess_output, pytorch_model_folder, mlflow_model_folder)
+
+    if is_main_process():
+        cleanup(completion_files_folder, model_selector_output,
+                preprocess_output, pytorch_model_folder, mlflow_model_folder)
+
+
+def parse_to_int(s):
+    """
+    To parse string to integer with default value in case of failure as one.
+
+    s: String which need to be parsed to integer.
+    """
+    try:
+        return int(s)
+    except ValueError:
+        return 1
 
 
 def _initiate_run(completion_files_folder: str, model_selector_output: str,
                   preprocess_output: str, pytorch_model_folder: str, mlflow_model_folder: str):
     """Run the model selector, preprocess, finetune and registration script."""
     # get task name
     task_name = decode_param_from_env_var("task_name")
+    nodes = parse_to_int(decode_param_from_env_var("Node_Count"))
+    gpus = parse_to_int(decode_param_from_env_var("number_of_gpu_to_use_finetuning"))

```

Comment:
who sets this number of gpus environment variable? And is it guaranteed that singularity will definitely start these many processes? Could it happen that it starts less than these in which case the barrier would never be released.

Reply:
> who sets this number of gpus environment variable? And is it guaranteed that singularity will definitely start these many processes? Could it happen that it starts less than these in which case the barrier would never be released.

Process count will be same as GPU count. This is tested . It's behavior would not change dynamically for every run. yes , barrier need to know how many process to wait , even in case of torch dist. this will be know internally based on group size